cpu/esp_common/Makefile.include

```diff
@@ -91,7 +97,10 @@ LINKFLAGS += -nostdlib -Wl,-gc-sections -Wl,-static
 ifeq (,$(filter esp_idf_heap,$(USEMODULE)))
   # use the wrapper functions for calloc to add correct overflow detection missing
   # in the newlib's version.
-  LINKFLAGS += -Wl,-wrap=_calloc_r
+  LINKFLAGS += -Wl,-wrap,_malloc_r
+  LINKFLAGS += -Wl,-wrap,_calloc_r
+  LINKFLAGS += -Wl,-wrap,_realloc_r
+  LINKFLAGS += -Wl,-wrap,_free_r
```

Comment:
```suggestion
  LINKFLAGS += -Wl,-wrap,_calloc_r
endif

ifneq (,$(filter esp_idf_heap,$(USEMODULE)))
  LINKFLAGS += -Wl,-wrap,_malloc_r
  LINKFLAGS += -Wl,-wrap,_calloc_r
  LINKFLAGS += -Wl,-wrap,_realloc_r
  LINKFLAGS += -Wl,-wrap,_free_r
```
You moved this block from `Makefile.dep` to here. The conditional that was already here was `ifeq` but the conditional in `Makefile.dep` was `ifneq`. That is, `_calloc_r` was wrapped in any case, but the others only if moculde `esp_idf_heap` is used.

So the following should work too:
```diff
--- a/cpu/esp_common/Makefile.include
+++ b/cpu/esp_common/Makefile.include
@@ -94,11 +94,12 @@ LINKFLAGS += -L$(ESP_SDK_DIR)/components/$(CPU)/lib
 endif
 LINKFLAGS += -nostdlib -Wl,-gc-sections -Wl,-static
 
-ifeq (,$(filter esp_idf_heap,$(USEMODULE)))
-  # use the wrapper functions for calloc to add correct overflow detection missing
-  # in the newlib's version.
+# use the wrapper functions for calloc to add correct overflow detection missing
+# in the newlib's version.
+LINKFLAGS += -Wl,-wrap,_calloc_r
+
+ifneq (,$(filter esp_idf_heap,$(USEMODULE)))
   LINKFLAGS += -Wl,-wrap,_malloc_r
-  LINKFLAGS += -Wl,-wrap,_calloc_r
   LINKFLAGS += -Wl,-wrap,_realloc_r
   LINKFLAGS += -Wl,-wrap,_free_r
 endif
```

Reply:
Wow, thanks I completely misread that conditional. I applied your suggestion in d96e4bab5f1ece133799e241f34240cff9c7467b

Reply:
The overflow detection was added by @maribu with PR #16443 :+1:. That's why `_calloc_r` wrapper function is used in any case and why I didn't realize the wrong conditional when reviewing the code :sunglasses: